test/fixtures.spec.js

```diff
@@ -16,13 +16,56 @@
  */
 
 const path = require('path');
-const {spawn} = require('child_process');
+const {spawn, execSync} = require('child_process');
 
-module.exports.describe = function({testRunner, expect, product, playwrightPath, FFOX, CHROMIUM, WEBKIT}) {
+module.exports.describe = function({testRunner, expect, product, playwright, playwrightPath, defaultBrowserOptions, WIN, FFOX, CHROMIUM, WEBKIT}) {
   const {describe, xdescribe, fdescribe} = testRunner;
   const {it, fit, xit, dit} = testRunner;
   const {beforeAll, beforeEach, afterAll, afterEach} = testRunner;
 
+  async function testSignal(action) {
+    const options = Object.assign({}, defaultBrowserOptions, {
+      // Disable DUMPIO to cleanly read stdout.
+      dumpio: false,
+      webSocket: true,
+      handleSIGINT: true,
+      handleSIGTERM: true,
+      handleSIGHUP: true,
+    });
+    const res = spawn('node', [path.join(__dirname, 'fixtures', 'closeme.js'), playwrightPath, product, JSON.stringify(options)]);
+    let wsEndPointCallback;
+    const wsEndPointPromise = new Promise(x => wsEndPointCallback = x);
+    let output = '';
+    let browserExitCode = 'none';
+    let browserSignal = 'none';
+    let browserPid;
+    res.stdout.on('data', data => {
+      output += data;
+      // Uncomment to debug these tests.
+      // console.log(data.toString());
```

Comment:
For your consideration:
```suggestion
    if (defaultBrowserOptions.dumpio)
      console.log(data.toString());
```